internal/pkg/manifest/validate.go

```diff
@@ -0,0 +1,388 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+)
+
+var (
+	intRangeBandRegexp = regexp.MustCompile(`^(\d+)-(\d+)$`)
+)
+
+// Validate returns if LoadBalancedWebServiceConfig is configured correctly.
+func (l LoadBalancedWebServiceConfig) Validate() error {
+	var err error
+	if err = l.ImageConfig.Validate(); err != nil {
```

Comment:
hmmm 💭 I think this flow overall makes sense but I'm worried a little about unit testing.
Ideally we should be able to do this in the tests:
```go
lbws := &LoadBalancedWebServiceConfig{
   ImageConfig: mockValidate.EXPECT().Validate().Return(errors.New("some error"),
}

require.EqualError(t, lbws.Validate(), `validate "image": some error`) 
```
i.e. the unit tests for the parent field should only test if we are wrapping the errors with the field name. The leaf field unit tests should validate the actual error messages.

Currently, the tests are too coupled. If we change `ImageConfig.Validate` then we have to also update every single parent unit tests.

How can we get to decouple them? Maybe there is no way but we should at least in our unit tests only test `validate "image"` and not the sub error does that make sense?

Reply:
OK done!